File: app/src/main/java/com/lugo/manueln/socialapp/Adapters/AdapterProfilePhotos.java
package com.lugo.manueln.socialapp.Adapters;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.lugo.manueln.socialapp.R;
import com.lugo.manueln.socialapp.objects.Post;
import com.squareup.picasso.Picasso;

import java.util.List;

public class AdapterProfilePhotos extends RecyclerView.Adapter<AdapterProfilePhotos.ViewProfilePhotos> {

    List<Post> listPreviewPhoto;

    public AdapterProfilePhotos(List<Post> listPreviewPhoto) {
        this.listPreviewPhoto = listPreviewPhoto;
    }

    @NonNull
    @Override
    public ViewProfilePhotos onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {

        View inflate=LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.preview_photo_item,viewGroup,false);

        return new ViewProfilePhotos(inflate);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewProfilePhotos viewProfilePhotos, int i) {


        Picasso.get()
                .load(listPreviewPhoto.get(i).getImage()).resize(960,960)
                .into(viewProfilePhotos.previewImagePost);



    }

    @Override
    public int getItemCount() {
        return listPreviewPhoto.size();
    }

    public class ViewProfilePhotos extends RecyclerView.ViewHolder {
        ImageView previewImagePost;
        public ViewProfilePhotos(@NonNull View itemView) {
            super(itemView);

            previewImagePost=itemView.findViewById(R.id.imagePreview);


        }
    }
}

File: app/src/main/java/com/lugo/manueln/socialapp/presentation/Adapters/AdapterProfilePhotos.kt
package com.lugo.manueln.socialapp.presentation.Adapters

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView

import com.lugo.manueln.socialapp.R
import com.lugo.manueln.socialapp.domain.Post
import com.squareup.picasso.Picasso

class AdapterProfilePhotos(internal var listPreviewPhoto: List<Post>) : RecyclerView.Adapter<AdapterProfilePhotos.ViewProfilePhotos>() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewProfilePhotos {

        val inflate = LayoutInflater.from(viewGroup.context).inflate(R.layout.preview_photo_item, viewGroup, false)

        return ViewProfilePhotos(inflate)
    }

    override fun onBindViewHolder(viewProfilePhotos: ViewProfilePhotos, i: Int) {


        Picasso.get()
                .load(listPreviewPhoto[i].image).resize(960, 960)
                .into(viewProfilePhotos.previewImagePost)


    }

    override fun getItemCount(): Int {
        return listPreviewPhoto.size
    }

    inner class ViewProfilePhotos(itemView: View) : RecyclerView.ViewHolder(itemView) {
        internal var previewImagePost: ImageView

        init {

            previewImagePost = itemView.findViewById(R.id.imagePreview)


        }
    }
}
